### ISSUE TITLE
Xcode shows an error message when importing a module that declares a typealias to a protocol with a primary associated type

### ISSUE THREAD

**Issue by tgrapperon** [apple/swift#60550]
**Describe the bug**
Xcode 14beta5 shows a message stating that "An internal error occurred. Source editor functionality is limited. Attempting to restore..." when importing a module that declare a typealias to a protocol with a primary associated type (and where the alias specifies the primary associated type, generically or not).

**Steps To Reproduce**
1. In the module `A` of a package, declare
```swift
public protocol Foo<Bar> {
  associatedtype Bar
}
public typealias Problem = Foo<String>
```
2. In a module `B`, import A, and then start to type `struct Hello {}`
3. It builds after completing the declaration, but it shows an error message at the top of the editor.

**Expected behavior**
Nothing happens.

**Screenshots**
<img width="868" alt="ErrorMessage" src="https://user-images.githubusercontent.com/35562418/184539314-c5cd7f99-f33c-4f7a-83d4-7fdef03cc830.png">

**Environment (please fill out the following information)**
 - macOS 13.0 beta 5 (22A5321d)
 - Xcode 4.0 beta 5 (14A5294e)

**Additional context**
Here is a package where the issue can be reproduced by starting editing the `EditThis.swift` file: [XcodeIssue.zip](https://github.com/apple/swift/files/9333446/XcodeIssue.zip).
This is likely related to https://github.com/pointfreeco/swift-composable-architecture/discussions/1264
Associated Feedback: FB11244375
The issue started to occur in Xcode 14.0 beta 4. There is no issue when using Xcode 14.0 beta 3.

**Comment by ahoppen** [comment]
Sorry for getting to this bug report so late. 

The problem should be fixed by https://github.com/apple/swift/pull/60801, which is included in the Xcode 14.1 betas. Could you check if you are still seeing the issue with these betas? If this continues to be an issue, feel free to re-open the issue and I’ll take another look.

**Comment by tgrapperon** [comment]
It indeed appeared to be fixed in 14.1b3, but I wasn't sure it was not coincidental so I planned to wait for the RC to confirm and eventually close it. Thanks for the update!

**Comment by ahoppen** [comment]
Thanks for checking.
